src/cephadm/cephadm

```diff
@@ -5212,6 +5222,185 @@ def command_gather_facts():
 ##################################
 
 
+class CephadmDaemonHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        # return the current state of the cache
+        self.send_response(200)
+        self.send_header('Content-type','application/json')
+        self.end_headers()
+        self.wfile.write(bytes(json.dumps(cephadm_cache), 'ascii'))
+    
+    def log_message(self, format, *args):
+        rqst = " ".join(args)
+        logger.info(f"client:{self.address_string()} [{self.log_date_time_string()}] {rqst}")
+
+
+class CephadmHTTPServer(ThreadingMixIn, HTTPServer):
+    allow_reuse_address = True
+    daemon_threads = True
+
+
+def cephadmd_install():
+    # open firewall port
+    # copy self to /usr/bin somewhere
+    # create the systemd unit file
+    # enable the systemd unit file
+    # start the systemd unit
+    pass
+
+def cephadmd_uninstall():
+    # shutdown the systemd unit
+    # disable it and remove it
+    # close the firewall port
+    pass
+
+def cephadmd_upgrade():
+    # copy self to the /usr/bin
+    # restart the systemd unit
+    pass
+
+
+class CephadmDaemon():
+    def __init__(self, fsid, port=None):
+        self.fsid = fsid
+        self.port = port
+        self.workers = []
+        self.http_server = None
+        self.stop = False
+
+    @property
+    def port_active(self):
+        return port_in_use(self.port)
+
+    @property
+    def can_run(self):
+        return not self.port_active
+    
+    def _scrape_host_facts(self, refresh_interval=10):
+        loop_delay = 0.1
```

Comment:
`0.1` is close to a busy loop. Is `1s` ok for you?

```suggestion
        loop_delay = 1
```

Reply:
loop delays are set at 1s - they're also a class attribute